test/libsyncengine/reconciliation/conflict_resolver/testconflictresolverworker.cpp

```diff
@@ -726,12 +726,62 @@ void TestConflictResolverWorker::testMoveMoveCycle() {
 
     _syncPal->_conflictResolverWorker->execute();
 
-    CPPUNIT_ASSERT(_syncPal->_syncOps->size() == 1);
+    CPPUNIT_ASSERT_EQUAL(size_t(1), _syncPal->_syncOps->size());
     UniqueId opId = _syncPal->_syncOps->opSortedList().front();
     SyncOpPtr op = _syncPal->_syncOps->getOp(opId);
-    CPPUNIT_ASSERT(op->newParentNode() == lNodeA);
-    CPPUNIT_ASSERT(op->targetSide() == ReplicaSideLocal);
-    CPPUNIT_ASSERT(op->type() == OperationTypeMove);
+    CPPUNIT_ASSERT_EQUAL(lNodeA, op->newParentNode());
+    CPPUNIT_ASSERT_EQUAL(ReplicaSideLocal, op->targetSide());
+    CPPUNIT_ASSERT_EQUAL(OperationTypeMove, op->type());
 }
 
+void TestConflictResolverWorker::testMoveMoveCycle2() {
+    /**
+     * Initial FS state:
+     *
+     *            root
+     *             |
+     *             A
+     *        _____|______
+     *       |           |
+     *      AA          AB
+     *      |
+     *     AAA
+     */
+
+    // Simulate move of node AAA to AB on local replica
+    std::shared_ptr<Node> lNodeA = _syncPal->_localUpdateTree->getNodeById("lA");
+    std::shared_ptr<Node> lNodeAA = _syncPal->_localUpdateTree->getNodeById("lAA");
+    std::shared_ptr<Node> lNodeAAA = _syncPal->_localUpdateTree->getNodeById("lAAA");
+    std::shared_ptr<Node> lNodeAB = _syncPal->_localUpdateTree->getNodeById("lAB");
+    lNodeAAA->setChangeEvents(OperationTypeMove);
+    lNodeAAA->setMoveOriginParentDbId(lNodeAA->idb());
+    lNodeAAA->setMoveOrigin("A/AA/AAA");
+    lNodeAAA->setParentNode(lNodeAB);
+    lNodeAA->deleteChildren(lNodeAAA);
+    lNodeAB->insertChildren(lNodeAAA);
+
+    // Simulate move of node AB to AA, on remote replica
```

Comment:
Are you sure this creates a cycle? UpdateTreeWorker::getNewPathAfterMove

Reply:
Typo in the comment AB is move to AAA not AA

Reply:
https://github.com/Infomaniak/desktop-kDrive/pull/203/files#r1677889007